### PR TITLE
Use Wayland for supported GUI packages

### PR DESCRIPTION
**Pull request by supechicken**
`handbrake` doesn't work with `#{CREW_OPTIONS}`

### CODE REVIEW

File: packages/homebank.rb [hunk 1]
```diff
@@ -3,35 +3,30 @@
...
-           "--prefix=#{CREW_PREFIX}",
-           "--libdir=#{CREW_LIB_PREFIX}"
+    system "./configure #{CREW_OPTIONS}"
     system 'make'
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/homebank', '#{CREW_DEST_PREFIX}/bin/homebank_orig'
+    system "cat <<'EOF'> homebank
+WAYLAND_DISPLAY=wayland-0
+GDK_BACKEND=wayland
+DISPLAY=
+#{CREW_PREFIX}/bin/homebank_orig $@
```
Comment (uberhacker) [inline packages/homebank.rb:27]: Put quotes around `\"$@\"`?

File: packages/handbrake.rb [hunk 2]
```diff
@@ -40,6 +35,14 @@ def self.install
     Dir.chdir 'build' do
       system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
     end
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/ghb', '#{CREW_DEST_PREFIX}/bin/ghb_orig'
+    system "cat <<'EOF'> ghb
+WAYLAND_DISPLAY=wayland-0
+GDK_BACKEND=wayland
+DISPLAY=
+#{CREW_PREFIX}/bin/ghb_orig $@
```
Comment (uberhacker) [inline packages/handbrake.rb:43]: Put quotes around `\"$@\"`?

File: packages/filezilla.rb [hunk 3]
```diff
@@ -46,5 +35,13 @@ def self.build
 
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/filezilla', '#{CREW_DEST_PREFIX}/bin/filezilla_orig'
+    system "cat <<'EOF'> filezilla
+WAYLAND_DISPLAY=wayland-0
+GDK_BACKEND=wayland
+DISPLAY=
+#{CREW_PREFIX}/bin/filezilla_orig $@
```
Comment (uberhacker) [inline packages/filezilla.rb:43]: Put quotes around `\"$@\"`?

File: packages/bluefish.rb [hunk 4]
```diff
@@ -29,18 +16,25 @@ class Bluefish < Package
...
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/bluefish', '#{CREW_DEST_PREFIX}/bin/bluefish_orig'
+    system "cat <<'EOF'> bluefish
+WAYLAND_DISPLAY=wayland-0
+GDK_BACKEND=wayland
+DISPLAY=
+#{CREW_PREFIX}/bin/bluefish_orig $@
```
Comment (uberhacker) [inline packages/bluefish.rb:35]: Put quotes around `\"$@\"`?

File: packages/homebank.rb [hunk 5]
```diff
@@ -3,35 +3,30 @@
...
+  depends_on 'xcb_util'
 
   def self.build
-    system './configure',
-           "--prefix=#{CREW_PREFIX}",
-           "--libdir=#{CREW_LIB_PREFIX}"
+    system "./configure #{CREW_OPTIONS}"
     system 'make'
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/homebank', '#{CREW_DEST_PREFIX}/bin/homebank_orig'
+    system "cat <<'EOF'> homebank
```
Comment (uberhacker) [inline packages/homebank.rb:23]: Missing `#!/bin/bash`?

File: packages/handbrake.rb [hunk 6]
```diff
@@ -40,6 +35,14 @@ def self.install
     Dir.chdir 'build' do
       system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
     end
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/ghb', '#{CREW_DEST_PREFIX}/bin/ghb_orig'
+    system "cat <<'EOF'> ghb
```
Comment (uberhacker) [inline packages/handbrake.rb:39]: Missing `#!/bin/bash`?

File: packages/filezilla.rb [hunk 7]
```diff
@@ -46,5 +35,13 @@ def self.build
 
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/filezilla', '#{CREW_DEST_PREFIX}/bin/filezilla_orig'
+    system "cat <<'EOF'> filezilla
```
Comment (uberhacker) [inline packages/filezilla.rb:39]: Missing `#!/bin/bash`?

File: packages/bluefish.rb [hunk 8]
```diff
@@ -29,18 +16,25 @@ class Bluefish < Package
...
-           "--libdir=#{CREW_LIB_PREFIX}",
-           '--disable-maintainer-mode'
+    system "./configure #{CREW_OPTIONS}"
     system "make"
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    FileUtils.mv '#{CREW_DEST_PREFIX}/bin/bluefish', '#{CREW_DEST_PREFIX}/bin/bluefish_orig'
+    system "cat <<'EOF'> bluefish
```
Comment (uberhacker) [inline packages/bluefish.rb:31]: Missing `#!/bin/bash`?

File: packages/bluefish.rb [hunk 9]
```diff
@@ -29,18 +16,25 @@ class Bluefish < Package
...
+WAYLAND_DISPLAY=wayland-0
+GDK_BACKEND=wayland
+DISPLAY=
```
Comment (uberhacker) [inline packages/bluefish.rb:34]: I wonder if we can put these variables in a `.env` file instead?

Comment (supechicken) [reply packages/bluefish.rb:34]: Can we put the `env` file into `install.sh`?

Comment (uberhacker) [reply packages/bluefish.rb:34]: We might be able to do something without the need of the shell script.

**Comment by supechicken** [comment]
Using `alias` might be a better solution :)

File: packages/handbrake.rb [hunk 10]
```diff
@@ -48,5 +43,8 @@ def self.postinstall
     puts
     puts "Type 'HandBrakeCLI' for the command line.".lightblue
     puts
+    puts "To complete the installation, execute the following:".lightblue
+    puts "echo 'alias bluefish=\"WAYLAND_DISPLAY=wayland-0 DISPLAY=\'\' GDK_BACKEND=wayland ghb\"' >> ~/.bashrc".lightblue
```
Comment (uberhacker) [inline packages/handbrake.rb:47]: `'alias ghb=...`?

File: packages/filezilla.rb [hunk 11]
```diff
@@ -47,4 +36,9 @@ def self.build
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
   end
+  def self.postinstall
+    puts
+    puts "To complete the installation, execute the following:".lightblue
+    puts "echo 'alias bluefish=\"WAYLAND_DISPLAY=wayland-0 DISPLAY=\'\' GDK_BACKEND=wayland filezilla\"' >> ~/.bashrc".lightblue
```
Comment (uberhacker) [inline packages/filezilla.rb:42]: `'alias filezilla=`?

File: packages/homebank.rb [hunk 12]
```diff
@@ -3,38 +3,30 @@
...
     system 'make'
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
   end
+  def self.postinstall
+    puts
+    puts "To complete the installation, execute the following:".lightblue
+    puts "echo 'alias bluefish=\"WAYLAND_DISPLAY=wayland-0 DISPLAY=\'\' GDK_BACKEND=wayland homebank\"' >> ~/.bashrc".lightblue
```
Comment (uberhacker) [inline packages/homebank.rb:30]: `'alias homebank=...`?

Comment (supechicken) [reply packages/homebank.rb]: Done

**Pull request merged**